### PR TITLE
Sync with upstream `v1.28.0`

### LINKED ISSUE #246: Sync with Upstream v1.28.0
## What would you like to be added
Gardener autoscaler should be synced with respective [Kubernetes Autoscaler 1.28.9 Release](https://github.com/kubernetes/autoscaler/releases/)

## Why is this needed

* To keep the fork in sync with upstream.
* To ensure the shoot control planes are using respective version of CA for a given K8S version

## Steps

Please follow the steps in
[How do I sync gardener autoscaler with an upstream autoscaler minor release](https://github.com/gardener/autoscaler/blob/machine-controller-manager-provider/cluster-autoscaler/FAQ.md#how-do-i-sync-gardener-autoscaler-with-an-upstream-autoscaler-minor-release)

### PR DESCRIPTION
**Pull request by aaronfern**
**What this PR does / why we need it**:
Sync with upstream autoscaler `v1.28.0`

**Which issue(s) this PR fixes**:
Fixes #246 

**Special notes for your reviewer**:

**Release note**:

```other operator
Synced changes till v1.28.0 of upstream autoscaler
```

### CODE REVIEW

File: cluster-autoscaler/core/static_autoscaler.go [hunk 1]
```diff
@@ -434,25 +434,18 @@ func (a *StaticAutoscaler) RunOnce(currentTime time.Time) caerrors.AutoscalerErr
 	// Check if there has been a constant difference between the number of nodes in k8s and
 	// the number of nodes on the cloud provider side.
 	// TODO: andrewskim - add protection for ready AWS nodes.
-
-	// FORK-CHANGE: Commented this code as it removes `Registered but long not Ready` nodes which causes issues like scaling below minimum size and removing ready nodes during meltdown scenario
```
Comment (rishabh-11) [inline cluster-autoscaler/core/static_autoscaler.go:438]: Never undo the `FORK-CHANGE`. Can you please revert this? These are the changes done in the core CA by us. We need to keep them. Please revert the changes at other similar places

**Review by rishabh-11 — CHANGES_REQUESTED** [review]

File: cluster-autoscaler/core/static_autoscaler.go (hunk 1 above)
Comment (elankath) [reply cluster-autoscaler/core/static_autoscaler.go:438]: We probably should add a custom linter for this and add it to run during `make verify` target.

File: cluster-autoscaler/core/scaleup/orchestrator/orchestrator.go [hunk 2]
```diff
@@ -431,55 +417,141 @@ func (o *ScaleUpOrchestrator) ScaleUpToNodeGroupMinSize(
...
+		}
+
+		currentTargetSize, err := nodeGroup.TargetSize()
+		if err != nil {
+			klog.Errorf("Failed to get node group size: %v", err)
+			skippedNodeGroups[nodeGroup.Id()] = NotReadyReason
+			continue
+		}
+		if currentTargetSize >= nodeGroup.MaxSize() {
+			klog.V(4).Infof("Skipping node group %s - max size reached", nodeGroup.Id())
```
Comment (rishabh-11) [inline cluster-autoscaler/core/scaleup/orchestrator/orchestrator.go]:
```suggestion
            // FORK-CHANGE: log level changed to 2 for better debugging.
			klog.V(2).Infof("Skipping node group %s - max size reached", nodeGroup.Id())
```
This is a fork change and should be kept as such.

**Review by rishabh-11 — CHANGES_REQUESTED** [review]
Small change needed. Rest looks good. You also need to add some details in the sync-changes md file (will let you know on slack) and need to run IT again before I merge.

File: cluster-autoscaler/core/scaleup/orchestrator/orchestrator.go (hunk 2 above)
Comment (aaronfern) [reply cluster-autoscaler/core/scaleup/orchestrator/orchestrator.go]: Thanks for pointing this
Updated

File: cluster-autoscaler/SYNC-CHANGES/SYNC_CHANGES-1.28.md [hunk 3]
```diff
@@ -17,6 +17,14 @@
 
 ## Changes made
  - See general release notes of 1.28.0: https://github.com/kubernetes/autoscaler/releases/tag/cluster-autoscaler-1.28.0
+ - New flag added: flag.String(config.SchedulerConfigFileFlag, "", "scheduler-config) allows changing configuration of in-tree scheduler plugins acting on PreFilter and Filter extension points")
+ - The followign options have been added per node group
```
Comment (rishabh-11) [inline cluster-autoscaler/SYNC-CHANGES/SYNC_CHANGES-1.28.md]:
```suggestion
 - The following options have been added per node group
```

**Review by rishabh-11 — CHANGES_REQUESTED** [review]
minor typo needs to be corrected. rest looks good

File: cluster-autoscaler/SYNC-CHANGES/SYNC_CHANGES-1.28.md (hunk 3 above)
Comment (aaronfern) [reply cluster-autoscaler/SYNC-CHANGES/SYNC_CHANGES-1.28.md]: Done

**Review by rishabh-11 — APPROVED** [review]
/lgtm
